srtcore/queue.cpp

```diff
@@ -216,24 +216,28 @@ CUnit *CUnitQueue::getNextAvailUnit()
     if (m_iCount >= m_iSize)
         return NULL;
 
-    CQEntry *entrance = m_pCurrQueue;
-
+    int units_checked = 0;
     do
     {
-        for (CUnit *sentinel = m_pCurrQueue->m_pUnit + m_pCurrQueue->m_iSize - 1; m_pAvailUnit != sentinel;
-             ++m_pAvailUnit)
+        CUnit *lastunit = m_pCurrQueue->m_pUnit + m_pCurrQueue->m_iSize - 1;
+        for (; m_pAvailUnit != lastunit; ++m_pAvailUnit, ++units_checked)
```

Comment:
This loop goes through every element in a queue except for the last one.

Reply:
AFAIR this is a circular list, so "sentinel" was the point from which the search started and it was expected to stop where done with one runaround. You just renamed it here to `lastunit`.